zerver/views/saved_snippets.py

```diff
@@ -0,0 +1,47 @@
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import gettext as _
+
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.response import json_success
+from zerver.lib.saved_snippets import (
+    do_create_saved_snippet,
+    do_delete_saved_snippet,
+    do_get_saved_snippets,
+)
+from zerver.lib.typed_endpoint import typed_endpoint
+from zerver.models import SavedSnippet, UserProfile
+
+
+def get_saved_snippets(
+    request: HttpRequest,
+    user_profile: UserProfile,
+) -> HttpResponse:
+    return json_success(request, data={"saved_snippets": do_get_saved_snippets(user_profile)})
+
+
+@typed_endpoint
+def create_saved_snippet(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    title: str,
+    content: str,
+) -> HttpResponse:
+    title = title.strip()[: SavedSnippet.MAX_TITLE_LENGTH]
+    content = content.strip()
+    if title == "":
+        raise JsonableError(_("Title cannot be empty."))
+    if content == "":
+        raise JsonableError(_("Content cannot be empty."))
```

Comment:
This should check for title being too long, and reject in that case. `content` should also be limited to `MAX_MESSAGE_LENGTH`. Use `StringConstraints` from Pydantic to do all the enforcement if possible rather than writing custom code like this, which in particular requires translators to translate additional strings.

Reply:
Also, posted https://chat.zulip.org/#narrow/stream/378-api-design/topic/saved.20replies/near/1942386 with some more questions.

Reply:
Changed `MAX_TITLE_LENGTH` to `60` and the request is rejected if the title's length exceeds that.